source/puppet/4.2/reference/release_notes.markdown

```diff
@@ -20,6 +20,25 @@ Make sure you also read the [Puppet 4.0 release notes](/puppet/4.0/reference/rel
 
 Also of interest: the [Puppet 4.1 release notes](/puppet/4.1/reference/release_notes.html).
 
+## Puppet 4.2.1
+
+Released July 21, 2015.
+
+Puppet 4.2.1 is a bug fix release. 
+
+* [PUP-4752](https://tickets.puppetlabs.com/browse/PUP-4752): Variable names were being checked for allowed characters, but not if they appeared as names of parameters.
+* [PUP-4770](https://tickets.puppetlabs.com/browse/PUP-4770): Solaris Zone's provider debug and error messages were changed to global, this has been reverted to so that it is prefixed with provider context.
+* [PUP-4777](https://tickets.puppetlabs.com/browse/PUP-4770): Puppet gem dependencies updated to use Hiera 3.
+* [PUP-4789](https://tickets.puppetlabs.com/browse/PUP-4789): The 4.x hiera_include function was not propagating the correct scope to the include function.
+* [PUP-4826](https://tickets.puppetlabs.com/browse/PUP-4826): When using Integer[1], the expectation was that this produces the type Integer[1, default], but it produced Integer[1,1]. The same was occuring with Float. After the fix, they are both operating as documented in the [Puppet Language Reference](http://docs.puppetlabs.com/puppet/4.2/reference/lang_data_number.html#parameters).
+* [PUP-4775](https://tickets.puppetlabs.com/browse/PUP-4775): Serialization of node objects could produce giant serializations. When loaded later would cause 'stack level too deep' errors. This was caused by logic missing in the node implementation that should have prevented serialization of the entire runtime state of the environment.
+* [PUP-4810](https://tickets.puppetlabs.com/browse/PUP-4810): Puppet was caching parse results even when environment_timeout was set to 0.
+* [PUP-4847](https://tickets.puppetlabs.com/browse/PUP-4847): When using custom providers, the puppet resource command could not use custom facts.
+
+* [All tickets addressed in 4.2.1](https://tickets.puppetlabs.com/issues/?filter=15117)
```

Comment:
Change to "All tickets fixed in" for consistency with entries below?

Also I think the Introduced in filter is good to go now. But let me know if not :)